### PR TITLE
strongswan: add hotplug script user handling

### PR DESCRIPTION
**Pull request by feckert**
Maintainer: @stintel
Compile tested: lede 17.01
Run tested: x86_64 apu3, lantiq_xrx200

Description:
Make strongswan to call scripts from /etc/hotplug.d/ipsec.

### CODE REVIEW

**Comment by stintel** [comment]
Please explain why this is needed in the commit message. Thanks.

**Comment by feckert** [comment]
@stintel ping any progress?

**Comment by stintel** [comment]
The commit message still doesn't say why it's needed. I am also a bit reluctant. Can we achieve the same thing without having to patch the default updown script?

@dedeckeh as a strongswan user, what do you think?

**Comment by feckert** [comment]
@stintel  There is already a patch in this repo which is patching this script.
See:
https://github.com/openwrt/packages/blob/master/net/strongswan/patches/300-include-ipsec-user-script.patch

I would replace this patch with my patch?
And update the commit message

**Comment by dedeckeh** [comment]
@stintel I'm not against the patch as it allows other packages to install ipsec hotplug scripts. As @feckert mentioned we already patch the _updown.in file.
But I would rename the patch to include-hotplug-ipsec.patch and update the comments in the file accordingly

File: net/strongswan/patches/320-change-user-hotplug-handling.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/src/_updown/_updown
++++ b/src/_updown/_updown
+@@ -18,7 +18,7 @@
+ 
+ # Add your custom ip rules to the /etc/ipsec.user file if you need that functionality.
```
Comment (dedeckeh) [inline net/strongswan/patches/320-change-user-hotplug-handling.patch]: Change the comment indicating a hotplug ipsec call is made now

File: net/strongswan/patches/320-change-user-hotplug-handling.patch [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+ 
+-[ -e /etc/ipsec.user ] && . /etc/ipsec.user "$1"
++[ -e /sbin/hotplug-call ] && /sbin/hotplug-call ipsec "$1"
```
Comment (dedeckeh) [inline net/strongswan/patches/320-change-user-hotplug-handling.patch]: No need to check hotplug-call is present as it's installed by the base-files package
Also use exec /sbin/hotplug-call ipsec "$1" (see https://git.openwrt.org/?p=openwrt/openwrt.git;a=blob;f=package/network/services/dnsmasq/files/dhcp-script.sh;h=86032694c156ba08539ca2513c9259dec47c9e46;hb=HEAD#l11)

**Comment by feckert** [comment]
@stintel @dedeckeh i have update my PR but i am not sure why i get a **The Travis CI build failed** ?
Do i have todo something?

**Comment by dedeckeh** [comment]
@feckert It looks like StrongSwan compilation is failing although it does not seem related to your changes

**Comment by feckert** [comment]
@stintel any progress on my PR?

File: net/strongswan/patches/320-change-user-hotplug-handling.patch (hunk 2 above)
Comment (sch-m) [reply net/strongswan/patches/320-change-user-hotplug-handling.patch]: @dedeckeh Using the exec system call will result in NOT executing the default content of the _updown script. Is this really what we want?

Comment (dedeckeh) [reply net/strongswan/patches/320-change-user-hotplug-handling.patch]: @sch-m No that's not what we want; see https://github.com/openwrt/packages/pull/7014

**Pull request merged**